### PR TITLE
[Shader Management] Foundry Multiple Shader Output (OLD)

### PR DESCRIPTION
**Pull request by cdxntchou**
Repos:
Trunk:  https://github.cds.internal.unity3d.com/unity/unity/pull/7609
Graphics:  https://github.com/Unity-Technologies/Graphics/pull/6824
Prototype:  https://github.cds.internal.unity3d.com/joshua-davis/BlockSurfaceShaders/pull/1

# **Please read the [Contributing guide](CONTRIBUTING.md) before making a PR.**

* Read the [Graphics repository & Yamato FAQ](http://go/graphics-yamato-faq).

### Checklist for PR maker
- [ ] Have you added a backport label (if needed)? For example, the `need-backport-*` label. After you backport the PR, the label changes to `backported-*`.
- [ ] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [ ] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR. If you do add documentation, make sure to add the relevant Graphics Docs team member as a reviewer of the PR. If you are not sure which person to add, see the [Docs team contacts sheet](https://docs.google.com/spreadsheets/d/1rgUWWgwLFEHIQ3Rz-LnK6PAKmbM49DZZ9al4hvnztOo/edit#gid=1058860420).
- [ ] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?

---
### Testing status
Describe what manual/automated tests were performed for this PR

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.shadergraph/Editor/ShaderFoundry/Samples/SimpleSampleBuilder.cs [hunk 1]
```diff
@@ -16,11 +16,12 @@ internal static void Build(ShaderContainer container, string shaderName, BuildCa
...
         {
+            // ideally the settings for this provider would be provided through the same interface as workflow settings
+            // but for now it is directly populated by this special constructor
             ITemplateProvider provider = new LegacyTemplateProvider(target, new ShaderGraph.AssetCollection());
 
-            var shaderInstBuilder = new ShaderInstance.Builder(container, shaderName);
-            
-            foreach(var template in provider.GetTemplates(container))
+            var shaderInstBuilder = new ShaderInstance.Builder(container, shaderName);            
+            foreach (var template in provider.GetTemplates(container))
```
Comment (joshua-davis) [inline com.unity.shadergraph/Editor/ShaderFoundry/Samples/SimpleSampleBuilder.cs:24]: Each template can go to a separate shader instance right? What's controlling this?

Comment (cdxntchou) [reply com.unity.shadergraph/Editor/ShaderFoundry/Samples/SimpleSampleBuilder.cs:51]: Fixed so SimpleShaderBuilder can enumerate and invoke additional shader ids

File: com.unity.shadergraph/Editor/ShaderFoundry/Samples/SimpleSample1.cs [hunk 2]
```diff
@@ -12,13 +12,14 @@ internal static void RunSimpleTest()
...
-            var shaderBuilder = new ShaderBuilder();
             SimpleSampleBuilder.BuildCommonTypes(container);
-            SimpleSampleBuilder.Build(container, target, ShaderName, BuildSample, shaderBuilder);
 
-            var code = shaderBuilder.ToString();
-            DisplayTestResult(ShaderName, code);
+            var primaryShaderID = string.Empty;
+            var generatedShader = SimpleSampleBuilder.Build(container, target, ShaderName, BuildSample, primaryShaderID);
+
+            DisplayTestResult(ShaderName, generatedShader.codeString);
```
[the remark this answers is not in the document]
Comment (cdxntchou) [reply com.unity.shadergraph/Editor/ShaderFoundry/Samples/SimpleSample1.cs:22]: I will change it to make it more future proof.  But yeah, in this specific test it is only building the primary shader, so generatedShader.shaderName == ShaderName (primary shader name).

**Review by joshua-davis — APPROVED** [review]
Looks good, but make sure to target `shader-foundry/staging`.

**Comment by cdxntchou** [comment]
Closing, new PR here https://github.com/Unity-Technologies/Graphics/pull/7056

**Pull request closed without merging**